### PR TITLE
📝 Rework Varnish doc

### LINKED ISSUE #2355: 📝 Clarify how to use Varnish
### Where on docs.platform.sh should be changed?

https://docs.platform.sh/configuration/services/varnish.html
https://docs.platform.sh/bestpractices/http-caching.html

### What exactly should be updated?

People have been confused about how to set up Varnish and how to purge the cache (see context).

Rework the page so it's clear what to do with Platform.sh. Don't get into too many details about Varnish itself, which should be covered by Varnish docs.

Update for:

-  Content accuracy
-  Includes what it should
-  Doesn't include what doesn't fit
-  Fits template
-  Fits style guide

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by CollierCZ**
## Why

Closes #2355

## What's changed

Reworked the main Varnish service page for style/conformance with the template.
Added some examples on how to do things.
Redid basic style of HTTP caching page and reworded references to Varnish (didn't go deep into the page, just tried to remove most things flagged by Vale).

### CODE REVIEW

**Review by trolologuy — CHANGES_REQUESTED** [review]
Great clarification of the pages!
The pros and cons of using varnish are now much clearer, as its usage.

- [ ] Final built pages are working as expected.
- [x] The information is correct.
- [x] The information is comprehensive.
- [x] The information is concise.
- [x] The information is clear.
- [x] If anything has been moved or removed, there are clear redirects to the new location or related content.
- [x] All links work, including to section headers.
- [x] The changes follow the content and markup style guides.
- [ ] The page fits the content model defined through the templates.

File: docs/src/add-services/varnish.md [hunk 1]
```diff
@@ -97,57 +127,114 @@ sub vcl_recv {
...
+               return(synth(403,"Not allowed."));
+           }
+           # Purge cache for allowed requests
+           return (purge);
+       }
+       ...
+   }
+   ```
+
+3. Purge cache with a curl call:
```
[the remark this answers is not in the document]
Comment (CollierCZ) [reply docs/src/add-services/varnish.md:233]: The idea was about an app being able to purge the cache with a push (like when it knows that it has updated content). That's why there's the link to the regions page with the IPs.

The cURL call is just an illustration so we don't have to include example from many different languages. It's simple and could be generalizable, I hope.

File: docs/src/add-services/varnish.md [hunk 2]
```diff
@@ -18,62 +19,89 @@ However, it is possible to configure a Varnish instance as part of an applicatio
...
-`<APP_NAME>` should match the name you gave your app in your [app configuration](../create-apps/app-reference.md).
+`<APP_NAME>` should match your app's `name` in the [app configuration](../create-apps/app-reference.md).
 
 The `configuration` block must reference a VCL file inside the `.platform` directory.
 The `path` defines the file relative to the `.platform` directory.
 
 {{% /endpoint-description %}}
 
-### 2. Create a VCL template file
+### 2. Create a VCL template
```
[the remark this answers is not in the document]
Comment (CollierCZ) [reply docs/src/add-services/varnish.md:48]: That breaks Varnish https://pr-2420-posreti-652soceglkw4u.eu-3.platformsh.site/add-services/varnish.html#circular-relationships

File: docs/src/add-services/varnish.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (trolologuy) [reply docs/src/add-services/varnish.md:233]: Okey that makes more sense. Should line 199 be changed from:
```suggestion
The following example shows how to set up purging.
```
to 
```suggestion
The following example shows how to set up purging from your app.
```
To make the app relationship clearer?

File: docs/src/add-services/varnish.md [hunk 3]
```diff
@@ -97,57 +127,114 @@ sub vcl_recv {
...
+  set req.backend_hint = application.backend();
+}
+```
+
+## Clear cache with a push
+
+You may want at times to clear a specific part of your cache when you know the content is outdated.
+With Varnish, you can clear the content with [purging and banning](https://varnish-cache.org/docs/trunk/users-guide/purging.html).
+
+The following example shows how to set up purging.
```
Comment (trolologuy) [inline docs/src/add-services/varnish.md:199]: :hammer_and_wrench:

Based on the previous comment, would the following:
```suggestion
The following example shows how to set up purging from your app.
```
Help make the context a bit clearer?

**Review by trolologuy — APPROVED** [review]
The files are now highlighted as expected.
There's an open discussion regarding context to the cache-purging, but it's non-blocking.

File: docs/src/add-services/varnish.md (hunk 3 above)
Comment (CollierCZ) [reply docs/src/add-services/varnish.md:199]: The example doesn't really show that. But I can add clarification.

**Review by trolologuy — APPROVED** [review]
The extra context makes it easier to understand the purpose and that extra steps are needed for implementation.
✅

**Review by trolologuy — APPROVED** [review]
✅

**Pull request merged**